headeremitter.js

```diff
@@ -728,26 +735,54 @@ HeaderEmitter.prototype.addDate = function (date) {
   if (date.getFullYear() < 1900 || date.getFullYear() > 9999)
     throw new Error("Date year is out of encodable range");
 
-  // Start by computing the timezone offset for a day. We lack a good format, so
-  // the the 0-padding is done by hand. Note that the tzoffset we output is in
-  // the form ±hhmm, so we need to separate the offset (in minutes) into an hour
-  // and minute pair.
-  let tzOffset = date.getTimezoneOffset();
-  let tzOffHours = Math.abs(Math.trunc(tzOffset / 60));
-  let tzOffMinutes = Math.abs(tzOffset) % 60;
-  let tzOffsetStr = (tzOffset > 0 ? "-" : "+") +
-    padTo2Digits(tzOffHours) + padTo2Digits(tzOffMinutes);
+  let day = "", dDate = "",  month = "", fullYear = "", hours = "", 
+      minutes = "", seconds = "", tzOffsetStr = ""
```

Comment:
It's simpler to replace most of this logic with something like:
```javascript
let getDayProp = this._useOnlyUTC ? (prop => date["getUTC" + prop]()) : (prop => date["get" + prop]());
```
and then you can say `getDayProp("Day")` without needing to have all of the getters in the block.